rbac/management/role/serializer.py

```diff
@@ -332,6 +332,11 @@ def validate_role_update(instance, validated_data):
         message = "System roles may not be updated."
         error = {key: [_(message)]}
         raise serializers.ValidationError(error)
+    if instance.admin_default:
```

Comment:
All `admin_default` roles should be `system` roles by default [via seeding](https://github.com/RedHatInsights/insights-rbac/blob/master/rbac/management/role/definer.py#L39), correct? We may not even need this check.

Reply:
I see where you are coming from. I guess I tend to lean towards explicitly checking for conditions, but if you think it would be redundant I can remove that check. Is there any likely hood of there being an `admin_default=true` but `system=false` role being created? Since these admin_default groups would be coming from the RBAC config repo, we would expect it to be `system=true`, is that the idea of why we don't need the check?

Reply:
Right, you can't supply the `admin_default` value (default false) or the `system` value (default false) via the API, so custom roles will never be `system=false` with `admin_default=true`. The only case this would happen is in the case where the business wants us to change to allow for "Custom Default Admin Access", similar to `platform_default`, in which case we'd flip the `system` flag to false when we update the group. But in that case, we'd still want the check as is, since you _should_ be able to update the group if it's no longer a `system` group. You make a good point about being explicit, though. In this case though it's really any `system` object which we want to protect.

Reply:
Ok that's a lot of good information and really clears it up for me. Thanks!